### PR TITLE
Stroke color function support added to customize bar borders based on value paramater

### PR DESCRIPTION
**Pull request by Digvijayrao-KF**
# New Pull Request

color prop under stroke object will now support function that will accept params like value, dataPointIndex & w. This will allow the user to change the border color based on values plotted on y-axis for horizontal bar charts.

Fixes #[4360](https://github.com/apexcharts/apexcharts.js/issues/4360)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] Existing unit tests pass locally with my changes

### CODE REVIEW

File: src/charts/Bar.js [hunk 1]
```diff
@@ -319,10 +319,38 @@ class Bar {
...
+     // Note: the position of this function logic (ex. stroke: { colors: ["",function(){}] }) i.e array index 1 depicts the realIndex/seriesIndex.
+      function fetchColor(i) {
+        const exp = w.config.stroke.colors
+        let c
+        if (Array.isArray(exp) && exp.length > 0) {
+          c = exp[i]
+          if (!c) c = ''
+          if (typeof c === 'function') {
+            return c({
+              value: w.globals.axisCharts
```
Comment (junedchhipa) [inline src/charts/Bar.js:332]: w.globals.axisCharts will always be true for bar chart, so no point in having this condition here.

File: src/charts/Bar.js [hunk 2]
```diff
@@ -319,10 +319,38 @@ class Bar {
...
+                ? w.globals.series[i][j]
+                  ? w.globals.series[i][j]
+                  : 0
+                : w.globals.series[i],
+              dataPointIndex: i,
```
Comment (junedchhipa) [inline src/charts/Bar.js:337]: dataPointIndex should be `j` not `i`

**Review by junedchhipa — CHANGES_REQUESTED** [review]

**Comment by junedchhipa** [comment]
@Digvijayrao-KF Can you post a screenshot of what this accomplishes? Also, post a code sample to how you will use this option so I can reproduce.

File: src/charts/Bar.js (hunk 1 above)
Comment (Digvijayrao-KF) [reply src/charts/Bar.js]: done

File: src/charts/Bar.js (hunk 2 above)
Comment (Digvijayrao-KF) [reply src/charts/Bar.js]: done

**Comment by Digvijayrao-KF** [comment]
> @Digvijayrao-KF Can you post a screenshot of what this accomplishes? Also, post a code sample to how you will use this option so I can reproduce.

Hi @junedchhipa ,

We can see the series 2 border color changes for the paths based on values plotted on inverse x-axis/y-axis

image

### Code Sample
> Below snippet can be added under react-apexcharts **example** directory for Bar.jsx file
> Usage is for stroke: { colors: [...] } prop

  ```
  this.state = {
      series: [
        {
          data: [ 4, 3.5, 5, 2.7, 3.8, 4.6 ]
        },
        {
          data: [ 5, 3.7, 2.8, 5, 3.2, 3.5 ]
        }
      ],
      options: {
        colors: [
          function ( { value, seriesIndex } )
          {
            if ( value < 3.6 ) return seriesIndex == 1 ? '#f3fbff' : '#54A0FF'
            else return seriesIndex == 1 ? '#f3fbff' : '#FFC759'
          }
        ],
        chart: {
          type: 'bar',
          height: 130
        },
        plotOptions: {
          bar: {
            borderRadius: 6,
            borderRadiusApplication: 'end',
            horizontal: true,
            barHeight: '50%'
          }
        },
        dataLabels: {
          enabled: false
        },
        xaxis: {
          categories: [
            'Optimism',
            'Trust',
            'Humility',
            'Influence',
            'Scalability',
            'Empathy'
          ],
          axisBorder: {
            offsetX: -3
          },
          min: 1,
          max: 5,
          stepSize: 1,
          forceNiceScale: true,
          tickAmount: 1
        },
        stroke: {
          show: true,
          width: 2,
          colors: [
            '',     // indicates series 1
            function ( { value } )    // indicates series 2
            {
              return value < 3.6 ? '#54A0FF' : '#FFC759'
            }
          ]
        },
        grid: {
          yaxis: {
            lines: {
              show: false
            }
          }
        },
        legend: {
          show: false
        }
      }
    }
```

**Comment by Digvijayrao-KF** [comment]
Hi @junedchhipa , hope to see your reply soon, as we need this feature to be available sooner. Thanks

**Review by junedchhipa — APPROVED** [review]

**Pull request merged**